std/file.d

```diff
@@ -1067,14 +1067,26 @@ private void removeImpl(scope const(char)[] name, scope const(FSChar)* namez) @t
         if (!name)
         {
             import core.stdc.string : strlen;
-            auto len = strlen(namez);
+
+            auto len = namez ? strlen(namez) : 0;
             name = namez[0 .. len];
         }
         cenforce(core.stdc.stdio.remove(namez) == 0,
             "Failed to remove file " ~ name);
     }
 }
 
+@safe unittest
+{
+    import std.exception : collectExceptionMsg;
+
+    string filename = null; // e.g. as returned by File.tmpfile.name
+    assert(
```

Comment:
This assert fails currently

Reply:
on linux it resulted in `Bad address` on my system, I adjusted the error messages now and test them on linux and windows. (message is platform-dependent) Otherwise just a check for FileException is used.

If the message is different on other linux distros (such as in the CI), I will also limit it to a prefix check there.